In [1]:
import sys, os
sys.path.insert(0, os.path.abspath('/home/ElasticNotebook'))
%load_ext ElasticNotebook
from elastic.core.common.pandas import compare_df
import pickle
import cudf

Enabled rmm statistics


<a id="1"></a>
# <p style="padding:10px;background-color:#A23434;margin:0;color:#CDCA34;font-family:newtimeroman;font-size:100%;text-align:center;border-radius: 15px 50px;overflow:hidden;font-weight:500">Importing Libraries</p>

In [2]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import os
# STEFANOS: Conditionally import Modin Pandas
if "IREWR_WITH_MODIN" in os.environ and os.environ["IREWR_WITH_MODIN"] == "True":
    # STEFANOS: Import Modin Pandas
    import os
    os.environ["MODIN_ENGINE"] = "ray"
    import ray
    ray.init(num_cpus=int(os.environ['MODIN_CPUS']), runtime_env={'env_vars': {'__MODIN_AUTOIMPORT_PANDAS__': '1'}})
    import modin.pandas as pd
else:
    # STEFANOS: Import regular Pandas
    import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import time
sns.set()

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

<a id="1"></a>
# <p style="padding:10px;background-color:#A23434;margin:0;color:#CDCA34;font-family:newtimeroman;font-size:100%;text-align:center;border-radius: 15px 50px;overflow:hidden;font-weight:500">Reading Dataset</p>

In [3]:
%%time
### cell 0 ###

df = pd.read_csv("/home/dias-benchmarks/notebooks/saisandeepjallepalli/adidas-retail-eda-data-visualization/input/adidas-us-retail-products-dataset/adidas.csv")

CPU times: user 23.3 ms, sys: 4.52 ms, total: 27.8 ms
Wall time: 27.1 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_post 0 ##
cudf_df['crawled_at'] = df['crawled_at']

# -- STEFANOS -- Replicate Data

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 1 ##
cudf_df['crawled_at'] = df['crawled_at']

In [4]:
%%time
### cell 1 ###

factor = 1000
df = pd.concat([df]*factor)
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 845000 entries, 0 to 844
Data columns (total 20 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   url             845000 non-null  object 
 1   name            845000 non-null  object 
 2   sku             845000 non-null  object 
 3   selling_price   845000 non-null  int64  
 4   original_price  829000 non-null  object 
 5   currency        845000 non-null  object 
 6   availability    845000 non-null  object 
 7   color           845000 non-null  object 
 8   category        845000 non-null  object 
 9   source          845000 non-null  object 
 10  source_website  845000 non-null  object 
 11  breadcrumbs     845000 non-null  object 
 12  description     845000 non-null  object 
 13  brand           845000 non-null  object 
 14  images          845000 non-null  object 
 15  country         845000 non-null  object 
 16  language        845000 non-null  object 
 17  average_rating  84

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_post 1 ##
cudf_df['crawled_at'] = df['crawled_at']

<a id="1"></a>
# <p style="padding:10px;background-color:#A23434;margin:0;color:#CDCA34;font-family:newtimeroman;font-size:100%;text-align:center;border-radius: 15px 50px;overflow:hidden;font-weight:500">Basic Information</p>

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 2 ##
cudf_df['crawled_at'] = df['crawled_at']

In [5]:
%%time
### cell 2 ###

# First 5 records in the DataFrame

df.head()

CPU times: user 269 µs, sys: 43 µs, total: 312 µs
Wall time: 241 µs


,url,name,sku,selling_price,original_price,currency,availability,color,category,source,source_website,breadcrumbs,description,brand,images,country,language,average_rating,reviews_count,crawled_at
0,https://www.adidas.com/us/beach-shorts/FJ5089....,Beach Shorts,FJ5089,40,NaN,USD,InStock,Black,Clothing,adidas United States,https://www.adidas.com,Women/Clothing,Splashing in the surf. Making memories with yo...,adidas,"https://assets.adidas.com/images/w_600,f_auto,...",USA,en,4.5,35,2021-10-23 17:50:17.331255
1,https://www.adidas.com/us/five-ten-kestrel-lac...,Five Ten Kestrel Lace Mountain Bike Shoes,BC0770,150,NaN,USD,InStock,Grey,Shoes,adidas United States,https://www.adidas.com,Women/Shoes,Lace up and get after it. The Five Ten Kestrel...,adidas,"https://assets.adidas.com/images/w_600,f_auto,...",USA,en,4.8,4,2021-10-23 17:50:17.423830
2,https://www.adidas.com/us/mexico-away-jersey/G...,Mexico Away Jersey,GC7946,70,NaN,USD,InStock,White,Clothing,adidas United States,https://www.adidas.com,Kids/Clothing,"Clean and crisp, this adidas Mexico Away Jerse...",adidas,"https://assets.adidas.com/images/w_600,f_auto,...",USA,en,4.9,42,2021-10-23 17:50:17.530834
3,https://www.adidas.com/us/five-ten-hiangle-pro...,Five Ten Hiangle Pro Competition Climbing Shoes,FV4744,160,NaN,USD,InStock,Black,Shoes,adidas United States,https://www.adidas.com,Five Ten/Shoes,The Hiangle Pro takes on the classic shape of ...,adidas,"https://assets.adidas.com/images/w_600,f_auto,...",USA,en,3.7,7,2021-10-23 17:50:17.615054
4,https://www.adidas.com/us/mesh-broken-stripe-p...,Mesh Broken-Stripe Polo Shirt,GM0239,65,NaN,USD,InStock,Blue,Clothing,adidas United States,https://www.adidas.com,Men/Clothing,Step up to the tee relaxed. This adidas golf p...,adidas,"https://assets.adidas.com/images/w_600,f_auto,...",USA,en,4.7,11,2021-10-23 17:50:17.702680


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 3 ##
cudf_df['crawled_at'] = df['crawled_at']

In [6]:
%%time
### cell 3 ###

# Checking for Null Values in the DataFrame

df.isna().sum()

# There 16 Null Values for Original_price Column

CPU times: user 582 ms, sys: 567 µs, total: 583 ms
Wall time: 580 ms


url                   0
name                  0
sku                   0
selling_price         0
original_price    16000
currency              0
availability          0
color                 0
category              0
source                0
source_website        0
breadcrumbs           0
description           0
brand                 0
images                0
country               0
language              0
average_rating        0
reviews_count         0
crawled_at            0
dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 4 ##
cudf_df['crawled_at'] = df['crawled_at']

In [7]:
%%time
### cell 4 ###

# Shape of the DataFrame

df.shape

CPU times: user 11 µs, sys: 2 µs, total: 13 µs
Wall time: 15.7 µs


(845000, 20)

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 5 ##
cudf_df['crawled_at'] = df['crawled_at']

In [8]:
%%time
### cell 5 ###

# Information about the DataFrame

df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 845000 entries, 0 to 844
Data columns (total 20 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   url             845000 non-null  object 
 1   name            845000 non-null  object 
 2   sku             845000 non-null  object 
 3   selling_price   845000 non-null  int64  
 4   original_price  829000 non-null  object 
 5   currency        845000 non-null  object 
 6   availability    845000 non-null  object 
 7   color           845000 non-null  object 
 8   category        845000 non-null  object 
 9   source          845000 non-null  object 
 10  source_website  845000 non-null  object 
 11  breadcrumbs     845000 non-null  object 
 12  description     845000 non-null  object 
 13  brand           845000 non-null  object 
 14  images          845000 non-null  object 
 15  country         845000 non-null  object 
 16  language        845000 non-null  object 
 17  average_rating  84

<a id="1"></a>
# <p style="padding:10px;background-color:#A23434;margin:0;color:#CDCA34;font-family:newtimeroman;font-size:100%;text-align:center;border-radius: 15px 50px;overflow:hidden;font-weight:500">Statistical Information</p>

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 6 ##
cudf_df['crawled_at'] = df['crawled_at']

In [9]:
%%time
### cell 6 ###

# Correlation Between Columns

df.corr(numeric_only=True)

CPU times: user 30.5 ms, sys: 4.03 ms, total: 34.6 ms
Wall time: 33.4 ms


,selling_price,average_rating,reviews_count
selling_price,1.000000,-0.213004,0.102654
average_rating,-0.213004,1.000000,0.023585
reviews_count,0.102654,0.023585,1.000000


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 7 ##
cudf_df['crawled_at'] = df['crawled_at']

In [10]:
%%time
### cell 7 ###

# Statistical informtion about the DataFrame

df.describe().T

CPU times: user 77.2 ms, sys: 12.4 ms, total: 89.7 ms
Wall time: 90.3 ms


,count,mean,std,min,25%,50%,75%,max
selling_price,845000.0,53.192899,31.393071,9.0,28.0,48.0,70.0,240.0
average_rating,845000.0,4.608402,0.293621,1.0,4.5,4.7,4.8,5.0
reviews_count,845000.0,426.178698,1228.431476,1.0,19.0,68.0,314.0,11750.0


<a id="1"></a>
# <p style="padding:10px;background-color:#A23434;margin:0;color:#CDCA34;font-family:newtimeroman;font-size:100%;text-align:center;border-radius: 15px 50px;overflow:hidden;font-weight:500">Feature Engineering</p>

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 8 ##
cudf_df['original_price'] = df['original_price']

In [11]:
%%time
### cell 8 ###

# The Percentage of Missing Values in the 'original_price' Column

100 * (df['original_price'].isna().sum() / len(df))

# As the percntage of Null Records is less than 5%, hence dropping the Null record Rows

CPU times: user 34.1 ms, sys: 286 µs, total: 34.4 ms
Wall time: 33.3 ms


1.8934911242603552

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['crawled_at'] = df['crawled_at']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 9 ##
cudf_df['country'] = df['country']

In [12]:
%%time
### cell 9 ###

# Dropping Null Values in the DataFrame

df.dropna(inplace=True, axis=0)

CPU times: user 649 ms, sys: 36.5 ms, total: 685 ms
Wall time: 682 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['crawled_at'] = df['crawled_at']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_post 9 ##
cudf_df['country'] = df['country']

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['url'] = df['url']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['sku'] = df['sku']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['currency'] = df['currency']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['source'] = df['source']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['source_website'] = df['source_website']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['description'] = df['description']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['brand'] = df['brand']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['images'] = df['images']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['country'] = df['country']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['language'] = df['language']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 10 ##
cudf_df['crawled_at'] = df['crawled_at']

In [13]:
%%time
### cell 10 ###

# Checking for Null Values in the DataFrame

df.isna().sum()

CPU times: user 565 ms, sys: 11.4 ms, total: 577 ms
Wall time: 572 ms


url               0
name              0
sku               0
selling_price     0
original_price    0
currency          0
availability      0
color             0
category          0
source            0
source_website    0
breadcrumbs       0
description       0
brand             0
images            0
country           0
language          0
average_rating    0
reviews_count     0
crawled_at        0
dtype: int64

In [14]:
%%time
### cell 11 ###

# Dropping 'currency' column as all records have 'USD' as currency
# Dropping 'source' column as all records have 'adidas United States' as value
# Dropping 'brand', 'country', 'language' columns as all records have same value

df.drop([ 'brand', 'country', 'language', 'source_website', 'images', 'crawled_at', 'url', 'sku', 'currency','source', 'description'], axis=1, inplace=True)

CPU times: user 50.2 ms, sys: 12.3 ms, total: 62.5 ms
Wall time: 61.2 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 12 ##
cudf_df['reviews_count'] = df['reviews_count']

In [15]:
%%time
### cell 12 ###

# First 5 records in the DataFrame

df.head()

CPU times: user 111 µs, sys: 0 ns, total: 111 µs
Wall time: 113 µs


,name,selling_price,original_price,availability,color,category,breadcrumbs,average_rating,reviews_count
15,Essentials Loose Logo Tank Top,20,$25,InStock,Purple,Clothing,Women/Clothing,4.8,116
16,Essentials Loose Logo Tank Top,20,$25,InStock,Pink,Clothing,Women/Clothing,4.8,116
18,Essentials Loose Logo Tank Top,20,$25,InStock,Green,Clothing,Women/Clothing,4.8,116
19,Formotion Sculpt Tights,48,$80,InStock,Blue,Clothing,Women/Clothing,4.2,144
20,Marvel X Ghosted.3 Firm Ground Cleats,64,$80,InStock,Blue,Shoes,Soccer/Shoes,4.4,160


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 13 ##
cudf_df['original_price'] = df['original_price']

In [16]:
%%time
### cell 13 ###

# Removing '$' from the DataFrame

df['original_price'] = df['original_price'].str.split('$')
df['original_price'] = df['original_price'].str[1]

CPU times: user 1.36 s, sys: 96.3 ms, total: 1.45 s
Wall time: 1.45 s


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 13 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 14 ##
cudf_df['reviews_count'] = df['reviews_count']

In [17]:
%%time
### cell 14 ###

# First 5 records in the DataFrame

df.head()

CPU times: user 310 µs, sys: 36 µs, total: 346 µs
Wall time: 275 µs


,name,selling_price,original_price,availability,color,category,breadcrumbs,average_rating,reviews_count
15,Essentials Loose Logo Tank Top,20,25,InStock,Purple,Clothing,Women/Clothing,4.8,116
16,Essentials Loose Logo Tank Top,20,25,InStock,Pink,Clothing,Women/Clothing,4.8,116
18,Essentials Loose Logo Tank Top,20,25,InStock,Green,Clothing,Women/Clothing,4.8,116
19,Formotion Sculpt Tights,48,80,InStock,Blue,Clothing,Women/Clothing,4.2,144
20,Marvel X Ghosted.3 Firm Ground Cleats,64,80,InStock,Blue,Shoes,Soccer/Shoes,4.4,160


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 15 ##
cudf_df['reviews_count'] = df['reviews_count']

In [18]:
%%time
### cell 15 ###

# Shape of the DataFrame

df.shape

CPU times: user 9 µs, sys: 1e+03 ns, total: 10 µs
Wall time: 12.6 µs


(829000, 9)

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['category'] = df['category']

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 16 ##
cudf_df['reviews_count'] = df['reviews_count']

In [19]:
%%time
### cell 16 ###

# Checking the Data Types for the columns

df.dtypes

CPU times: user 118 µs, sys: 13 µs, total: 131 µs
Wall time: 136 µs


name               object
selling_price       int64
original_price     object
availability       object
color              object
category           object
breadcrumbs        object
average_rating    float64
reviews_count       int64
dtype: object

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 17 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [20]:
%%time
### cell 17 ###

# Creating 'Category' Column

df['Category'] = df['breadcrumbs'].str.split("/")
df['Category'] = df['Category'].str[0]

CPU times: user 1.56 s, sys: 59.9 ms, total: 1.62 s
Wall time: 1.61 s


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 17 ##
cudf_df['Category'] = df['Category']

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 18 ##
cudf_df['breadcrumbs'] = df['breadcrumbs']

In [21]:
%%time
### cell 18 ###

# Creating 'Product Type' Column

df['Product_Type'] = df['breadcrumbs'].str.split("/")
df['Product_Type'] = df['Product_Type'].str[1]

CPU times: user 1.65 s, sys: 32.1 ms, total: 1.68 s
Wall time: 1.68 s


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 18 ##
cudf_df['Product_Type'] = df['Product_Type']

In [22]:
%%time
### cell 19 ###

# Droping 'breadcrumbs' Column

df.drop(['breadcrumbs', 'category'], axis=1, inplace=True)

CPU times: user 98.1 ms, sys: 20.2 ms, total: 118 ms
Wall time: 117 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 20 ##
cudf_df['original_price'] = df['original_price']

In [23]:
%%time
### cell 20 ###

# Changing the DataType of 'original_price' from object to int64

df['original_price'] = df['original_price'].astype('int64')

CPU times: user 76.3 ms, sys: 24 µs, total: 76.3 ms
Wall time: 75.3 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 20 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['Category'] = df['Category']

In [ ]:
%%time
## Transfer_pre 21 ##
cudf_df['Product_Type'] = df['Product_Type']

In [24]:
%%time
### cell 21 ###

# Checking the Data Types for the columns

df.dtypes

CPU times: user 176 µs, sys: 17 µs, total: 193 µs
Wall time: 196 µs


name               object
selling_price       int64
original_price      int64
availability       object
color              object
average_rating    float64
reviews_count       int64
Category           object
Product_Type       object
dtype: object

<a id="1"></a>
# <p style="padding:10px;background-color:#A23434;margin:0;color:#CDCA34;font-family:newtimeroman;font-size:100%;text-align:center;border-radius: 15px 50px;overflow:hidden;font-weight:500">EDA and Data Visualization</p>

# Heat-Map

In [25]:
# STEFANOS: Disable plotting
# sns.heatmap(df.corr(), annot=True, cmap='magma')
# plt.show();

# Distribution Plots

In [26]:
# Distribution Plot for selling price

# STEFANOS: Disable plotting
# sns.displot(df['selling_price'], kde=True);

# The Below Graph is 'Right Skewed' with Majority of Data falling between 10-70.

In [27]:
# Distribution Plot for origial price

# STEFANOS: Disable plotting
# sns.displot(df['original_price'], kde=True);

# The Below Graph is 'Right Skewed' with Majority of Data falling between 10-90.

In [28]:
# Distribution Plot for average rating

# STEFANOS: Disable plotting
# sns.displot(df['average_rating'], kde=True);

# The Below Graph is 'Left Skewed' with Majority of Data falling between 4.2-5.0.

In [29]:
# Distribution Plot for reviews count

# STEFANOS: Disable plotting
# sns.displot(df['reviews_count'], kde=True);

# The Below Graph is 'Right Skewed' with Majority of Data falling below 600.

# Product Name

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 22 ##
cudf_df['name'] = df['name']

In [30]:
%%time
### cell 22 ###

# No.of Unique Products in the DataFrame

df['name'].nunique()

CPU times: user 51.9 ms, sys: 7.62 ms, total: 59.5 ms
Wall time: 58.7 ms


417

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 23 ##
cudf_df['name'] = df['name']

In [31]:
%%time
### cell 23 ###

# Top 10 - Products sold in the DataFrame

df['name'].value_counts()[:10]

CPU times: user 38.2 ms, sys: 0 ns, total: 38.2 ms
Wall time: 37.6 ms


name
ZX 1K Boost Shoes          24000
ZX 2K Boost Shoes          18000
Superstar Shoes            15000
EQ21 Run Shoes             13000
Racer TR21 Shoes           12000
ZX 2K Boost 2.0 Shoes      10000
Supernova+ Shoes            9000
4D Fusio Shoes              9000
Adilette Comfort Slides     9000
Swift Run X Shoes           9000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 24 ##
cudf_df['name'] = df['name']

In [32]:
%%time
### cell 24 ###

# Top 5 - Products sold in the DataFrame


# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['name'].value_counts()[:5].plot(kind='barh', color={'#864879','#2D4263','#C84B31', '#ECDBBA', '#B3541E'})
# plt.ylabel("Product Name")
# plt.xlabel("No.of Units Sold")
# plt.title("Product Name Vs. No.of Units Sold")
# plt.show();
df['name'].value_counts()[:5]

CPU times: user 37.1 ms, sys: 101 µs, total: 37.2 ms
Wall time: 36.7 ms


name
ZX 1K Boost Shoes    24000
ZX 2K Boost Shoes    18000
Superstar Shoes      15000
EQ21 Run Shoes       13000
Racer TR21 Shoes     12000
Name: count, dtype: int64

#### The Product '**ZX 1K Boost Shoes**' is the most sold Product/Recorded in the DataFrame with '**24**' units.

# Selling Price

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 25 ##
cudf_df['selling_price'] = df['selling_price']

In [33]:
%%time
### cell 25 ###

# No.of Unique selling price in the DataFrame

df['selling_price'].nunique()

CPU times: user 5.18 ms, sys: 109 µs, total: 5.29 ms
Wall time: 4.96 ms


68

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 26 ##
cudf_df['selling_price'] = df['selling_price']

In [34]:
%%time
### cell 26 ###

# Top 15 - Products interms of similar selling price in the DataFrame

df['selling_price'].value_counts()[:15]

CPU times: user 4.8 ms, sys: 0 ns, total: 4.8 ms
Wall time: 4.68 ms


selling_price
56     54000
20     50000
52     42000
32     40000
36     39000
28     35000
72     30000
60     29000
24     29000
70     28000
44     26000
40     25000
64     22000
48     22000
105    21000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 27 ##
cudf_df['selling_price'] = df['selling_price']

In [35]:
%%time
### cell 27 ###

# Top 5 - Products interms of similar selling price in the DataFrame

# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['selling_price'].value_counts()[:5].plot(kind='barh', color={'#864879','#2D4263','#C84B31', '#ECDBBA', '#B3541E'})
# plt.ylabel("Product Selling Price ($USD)")
# plt.xlabel("No.of Products")
# plt.title("Product Selling Price ($USD) Vs. No.of Products")
# plt.show();
df['selling_price'].value_counts()[:5]

CPU times: user 4.3 ms, sys: 27 µs, total: 4.33 ms
Wall time: 4.25 ms


selling_price
56    54000
20    50000
52    42000
32    40000
36    39000
Name: count, dtype: int64

In [36]:
# Top 15 - Products interms of similar selling price in the DataFrame

# STEFANOS: Disable plotting
# labels = df['selling_price'].value_counts().head(15).index
# pie, ax = plt.subplots(figsize=[20,8])
# plt.pie(x=df['selling_price'].value_counts().head(15).values, autopct="%.1f%%", explode=[0.06]*15, labels=labels, pctdistance=0.5)
# plt.title("Top 15 - Products interms of similar selling price ($USD)", fontsize=14);

## Maximum Selling Price

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 28 ##
cudf_df['selling_price'] = df['selling_price']

In [37]:
%%time
### cell 28 ###

# Maximum Selling Price

print("The Maximun Selling Price is:",df['selling_price'].max(),"USD")

The Maximun Selling Price is: 240 USD
CPU times: user 924 µs, sys: 86 µs, total: 1.01 ms
Wall time: 575 µs


## Minimum Selling Price

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 29 ##
cudf_df['selling_price'] = df['selling_price']

In [38]:
%%time
### cell 29 ###

# Minimum Selling Price

print("The Minimum Selling Price is:",df['selling_price'].min(),"USD")

The Minimum Selling Price is: 9 USD
CPU times: user 1.06 ms, sys: 99 µs, total: 1.16 ms
Wall time: 572 µs


## Average Selling Price

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 30 ##
cudf_df['selling_price'] = df['selling_price']

In [39]:
%%time
### cell 30 ###

# Average Selling Price

print("The Average Selling Price is:",round(df['selling_price'].mean(),2),"USD")

The Average Selling Price is: 52.91 USD
CPU times: user 1.38 ms, sys: 0 ns, total: 1.38 ms
Wall time: 1.04 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 31 ##
cudf_df['selling_price'] = df['selling_price']

In [40]:
%%time
### cell 31 ###

# Top 10 - Highest Selling Prices in USD

set(df['selling_price'].sort_values(ascending=False)[:30].values)

CPU times: user 58.8 ms, sys: 0 ns, total: 58.8 ms
Wall time: 57.6 ms


{240}

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 32 ##
cudf_df['selling_price'] = df['selling_price']

In [41]:
%%time
### cell 32 ###

# Top 10 - Least Selling Prices in USD

set(df['selling_price'].sort_values()[:40].values)

CPU times: user 57.5 ms, sys: 0 ns, total: 57.5 ms
Wall time: 56.9 ms


{9}

#### Highest Selling price is '**240 USD**' and Least is '**9 USD**' and product(s) with selling price of '**56 USD**' is sold '**54**' times.

# Original Price

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 33 ##
cudf_df['original_price'] = df['original_price']

In [42]:
%%time
### cell 33 ###

# No.of Unique original price in the DataFrame

df['original_price'].nunique()

CPU times: user 6.23 ms, sys: 0 ns, total: 6.23 ms
Wall time: 4.97 ms


42

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 34 ##
cudf_df['original_price'] = df['original_price']

In [43]:
%%time
### cell 34 ###

# Top 15 - Products interms of similar original price in the DataFrame

df['original_price'].value_counts()[:15]

CPU times: user 4.52 ms, sys: 644 µs, total: 5.17 ms
Wall time: 4.94 ms


original_price
65     68000
25     57000
100    55000
45     55000
30     49000
90     48000
70     47000
80     46000
35     40000
120    35000
50     33000
60     32000
40     32000
55     32000
150    29000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 35 ##
cudf_df['original_price'] = df['original_price']

In [44]:
%%time
### cell 35 ###

# Top 5 - Products interms of similar origial price in the DataFrame

# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['original_price'].value_counts()[:5].plot(kind='barh', color={'#864879','#2D4263','#C84B31', '#ECDBBA', '#B3541E'})
# plt.ylabel("Original Product Price ($USD)")
# plt.xlabel("No.of Products")
# plt.title("Original Product Price ($USD) Vs. No.of Products")
# plt.show();
df['original_price'].value_counts()[:5]

CPU times: user 5.13 ms, sys: 50 µs, total: 5.18 ms
Wall time: 4.86 ms


original_price
65     68000
25     57000
100    55000
45     55000
30     49000
Name: count, dtype: int64

In [45]:
# Top 15 - Products interms of similar Original price in the DataFrame

# STEFANOS: Disable plotting
# labels = df['original_price'].value_counts().head(15).index
# pie, ax = plt.subplots(figsize=[20,8])
# plt.pie(x=df['original_price'].value_counts().head(15).values, autopct="%.1f%%", explode=[0.06]*15, labels=labels, pctdistance=0.5)
# plt.title("Top 15 - Products interms of similar original price ($USD)", fontsize=14);

## Maximum Original Price

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 36 ##
cudf_df['original_price'] = df['original_price']

In [46]:
%%time
### cell 36 ###

# Maximum Original Price

print("The Maximun Original Price is:",df['original_price'].max(),"USD")

The Maximun Original Price is: 300 USD
CPU times: user 1.14 ms, sys: 0 ns, total: 1.14 ms
Wall time: 745 µs


## Minimum Original Price

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 37 ##
cudf_df['original_price'] = df['original_price']

In [47]:
%%time
### cell 37 ###

# Minimum Original Price

print("The Minimum Original Price is:",df['original_price'].min(),"USD")

The Minimum Original Price is: 14 USD
CPU times: user 884 µs, sys: 0 ns, total: 884 µs
Wall time: 522 µs


## Average Original Price

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 38 ##
cudf_df['original_price'] = df['original_price']

In [48]:
%%time
### cell 38 ###

# Average Original Price

print("The Average Original Price is:",round(df['original_price'].mean(),2),"USD")

The Average Original Price is: 69.01 USD
CPU times: user 1.52 ms, sys: 0 ns, total: 1.52 ms
Wall time: 1.13 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 39 ##
cudf_df['original_price'] = df['original_price']

In [49]:
%%time
### cell 39 ###

# Top 10 - Highest Original Prices in USD

set(df['original_price'].sort_values(ascending=False)[:80].values)

CPU times: user 56.4 ms, sys: 28 µs, total: 56.5 ms
Wall time: 56.1 ms


{300}

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 40 ##
cudf_df['original_price'] = df['original_price']

In [50]:
%%time
### cell 40 ###

# Top 10 - Least Original Prices in USD

set(df['original_price'].sort_values().values[:100])

CPU times: user 54.1 ms, sys: 3.45 ms, total: 57.5 ms
Wall time: 56.8 ms


{14}

#### Highest Original price is '**300 USD**' and Least is '**14 USD**' and product(s) with selling price of '**65 USD**' is sold '**68**' times.

# Discount

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 41 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 41 ##
cudf_df['original_price'] = df['original_price']

In [51]:
%%time
### cell 41 ###

# Calculating Discount

df['Discount'] = df['original_price'] - df['selling_price']

CPU times: user 3.02 ms, sys: 269 µs, total: 3.29 ms
Wall time: 2.9 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 41 ##
cudf_df['Discount'] = df['Discount']

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 42 ##
cudf_df['Discount'] = df['Discount']

In [52]:
%%time
### cell 42 ###

# No.of Unique Discount in the DataFrame

df['Discount'].nunique()

CPU times: user 6.06 ms, sys: 0 ns, total: 6.06 ms
Wall time: 5.04 ms


41

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 43 ##
cudf_df['Discount'] = df['Discount']

In [53]:
%%time
### cell 43 ###

# Top 15 - Highest Discount Amount in the DataFrame

list(set(df['Discount'].unique()))[-15::]

CPU times: user 5.55 ms, sys: 0 ns, total: 5.55 ms
Wall time: 4.65 ms


[32, 33, 36, 37, 39, 40, 42, 45, 48, 50, 54, 60, 72, 75, 84]

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 44 ##
cudf_df['Discount'] = df['Discount']

In [54]:
%%time
### cell 44 ###

# Top 15 - Least Discount Amount in the DataFrame

list(set(df['Discount'].unique()))[:15]

CPU times: user 6.25 ms, sys: 0 ns, total: 6.25 ms
Wall time: 5.08 ms


[2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 45 ##
cudf_df['Discount'] = df['Discount']

In [55]:
%%time
### cell 45 ###

# Top 15 - Most Given Discount Amount in the DataFrame

df['Discount'].value_counts()[:15]

CPU times: user 5.05 ms, sys: 96 µs, total: 5.15 ms
Wall time: 4.88 ms


Discount
5     73000
9     61000
13    57000
10    40000
7     39000
14    38000
6     36000
18    33000
12    33000
8     32000
30    30000
15    29000
16    29000
4     28000
24    28000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 46 ##
cudf_df['Discount'] = df['Discount']

In [56]:
%%time
### cell 46 ###

# Top 5 - Most Given Discount Amount in the DataFrame

# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['Discount'].value_counts()[:5].plot(kind='barh', color={'#864879','#2D4263','#C84B31', '#ECDBBA', '#B3541E'})
# plt.ylabel("Discount Amount")
# plt.xlabel("No.of times Particular Disount Amount Given")
# plt.title("Discount Amount Vs. No.of times Particular Disount Amount Given")
# plt.show();
df['Discount'].value_counts()[:5]

CPU times: user 5.13 ms, sys: 104 µs, total: 5.23 ms
Wall time: 5.09 ms


Discount
5     73000
9     61000
13    57000
10    40000
7     39000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 47 ##
cudf_df['Discount'] = df['Discount']

In [57]:
%%time
### cell 47 ###

# Top 15 - Most Given Discount Amount in the DataFrame

labels = df['Discount'].value_counts().head(15).index
# STEFANOS: Disable plotting
# pie, ax = plt.subplots(figsize=[20,8])
# plt.pie(x=df['Discount'].value_counts().head(15).values, autopct="%.1f%%", explode=[0.06]*15, labels=labels, pctdistance=0.5)
# plt.title("Top 15 - Most Given Discount Amount in the DataFrame", fontsize=14);
_ = df['Discount'].value_counts().head(15).values

CPU times: user 9.65 ms, sys: 0 ns, total: 9.65 ms
Wall time: 9.33 ms


#### Highest Discount Amount is '**84 USD**' and Least is '**2 USD**' and product(s) with Disount Amount of '**5 USD**' is sold '**73**' times.

# Discount Percentage(%)

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 48 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 48 ##
cudf_df['original_price'] = df['original_price']

In [58]:
%%time
### cell 48 ###

# Calculating Discount percentage(%)

df['Discount(%)'] = round(((df['original_price'] - df['selling_price']) / (df['original_price']))*100,2)

CPU times: user 8.59 ms, sys: 52 µs, total: 8.64 ms
Wall time: 7.68 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_post 48 ##
cudf_df['Discount(%)'] = df['Discount(%)']

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 49 ##
cudf_df['Discount(%)'] = df['Discount(%)']

In [59]:
%%time
### cell 49 ###

# No.of Unique Discount Percentage(%) in the DataFrame

df['Discount(%)'].nunique()

CPU times: user 5.74 ms, sys: 3.43 ms, total: 9.17 ms
Wall time: 8.23 ms


40

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 50 ##
cudf_df['Discount(%)'] = df['Discount(%)']

In [60]:
%%time
### cell 50 ###

# Top 15 - Highest Discount Percentage(%) in the DataFrame

top_discount_percnet = list(set(df['Discount(%)'].unique()))
top_discount_percnet.sort(reverse=True)
print(top_discount_percnet[:15])

[50.0, 49.33, 49.23, 40.0, 39.53, 35.71, 30.0, 29.41, 29.33, 29.31, 29.23, 29.17, 29.09, 28.89, 28.85]
CPU times: user 8.71 ms, sys: 63 µs, total: 8.77 ms
Wall time: 7.81 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 51 ##
cudf_df['Discount(%)'] = df['Discount(%)']

In [61]:
%%time
### cell 51 ###

# Top 15 - Least Discount Percentage(%) in the DataFrame

least_discount_percnet = list(set(df['Discount(%)'].unique()))
least_discount_percnet.sort(reverse=False)
print(least_discount_percnet[:15])

[7.14, 8.0, 8.57, 8.89, 9.09, 9.23, 9.33, 10.0, 14.29, 16.67, 17.39, 17.86, 18.18, 18.6, 18.75]
CPU times: user 5.16 ms, sys: 3.75 ms, total: 8.91 ms
Wall time: 8.12 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 52 ##
cudf_df['Discount(%)'] = df['Discount(%)']

In [62]:
%%time
### cell 52 ###

# Top 15 - Most Given Discount Percentage(%) in the DataFrame

df['Discount(%)'].value_counts().head(15)

CPU times: user 8.81 ms, sys: 74 µs, total: 8.89 ms
Wall time: 8.8 ms


Discount(%)
20.00    454000
30.00    173000
10.00     44000
29.23     20000
28.57     15000
28.89     15000
40.00     11000
17.86     11000
27.78      8000
29.41      8000
28.00      7000
18.18      6000
29.33      5000
8.89       5000
7.14       4000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 53 ##
cudf_df['Discount(%)'] = df['Discount(%)']

In [63]:
%%time
### cell 53 ###

# Top 5 - Most Given Discount Percentage in the DataFrame

# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['Discount(%)'].value_counts().head(5).plot(kind='barh', color={'#864879','#2D4263','#C84B31', '#ECDBBA', '#B3541E'})
# plt.ylabel("Discount Percentage")
# plt.xlabel("No.of times Particular Disount Percentage Given")
# plt.title("Discount Percentage Vs. No.of times Particular Disount Percentage Given")
# plt.show();
df['Discount(%)'].value_counts().head(5)

CPU times: user 9.46 ms, sys: 133 µs, total: 9.59 ms
Wall time: 9.24 ms


Discount(%)
20.00    454000
30.00    173000
10.00     44000
29.23     20000
28.57     15000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 54 ##
cudf_df['Discount(%)'] = df['Discount(%)']

In [64]:
%%time
### cell 54 ###

# Top 15 - Most Given Discount Percentage in the DataFrame

labels = df['Discount(%)'].value_counts().head(15).index
# STEFANOS: Disable plotting
# pie, ax = plt.subplots(figsize=[20,15])
# plt.pie(x=df['Discount(%)'].value_counts().head(15).values, autopct="%.1f%%", explode=[0.06]*15, labels=labels, pctdistance=0.5)
# plt.title("Top 15 - Most Given Discount Percenatage in the DataFrame", fontsize=14);
# plt.tight_layout();
_=df['Discount(%)'].value_counts().head(15).values

CPU times: user 14.3 ms, sys: 3.81 ms, total: 18.1 ms
Wall time: 17.6 ms


#### Highest Discount% is '**50%**' and Least is '**7.14%**' and product(s) with Disount Percentage of '**20**' is sold '**254**' times.

# Color

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 55 ##
cudf_df['color'] = df['color']

In [65]:
%%time
### cell 55 ###

# Unique colors in the DataFrame

df['color'].unique()

CPU times: user 41 ms, sys: 84 µs, total: 41.1 ms
Wall time: 40.4 ms


array(['Purple', 'Pink', 'Green', 'Blue', 'Grey', 'White', 'Yellow',
       'Black', 'Red', 'Multicolor', 'Gold', 'Burgundy', 'Beige',
       'Orange', 'Multi', 'Turquoise', 'Silver', 'Brown'], dtype=object)

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 56 ##
cudf_df['color'] = df['color']

In [66]:
%%time
### cell 56 ###

# No.of Unique colors in the DataFrame

df['color'].nunique()

CPU times: user 43 ms, sys: 0 ns, total: 43 ms
Wall time: 42.4 ms


18

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 57 ##
cudf_df['color'] = df['color']

In [67]:
%%time
### cell 57 ###

# No.of Products for each Color

df['color'].value_counts()

CPU times: user 38 ms, sys: 0 ns, total: 38 ms
Wall time: 37.3 ms


color
White         218000
Black         179000
Blue          102000
Grey           79000
Pink           62000
Green          59000
Purple         31000
Red            25000
Multicolor     20000
Yellow         17000
Orange         11000
Burgundy        9000
Beige           6000
Multi           4000
Gold            3000
Turquoise       2000
Silver          1000
Brown           1000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 58 ##
cudf_df['color'] = df['color']

In [68]:
%%time
### cell 58 ###

# No.of Products for each Color

labels = df['color'].value_counts().index
# STEFANOS: Disable plotting
# pie, ax = plt.subplots(figsize=[20,15])
# plt.pie(x=df['color'].value_counts().values, autopct="%.1f%%", explode=[0.03]*18, labels=labels, pctdistance=0.5)
# plt.title("No.of Products for each Color in the DataFrame", fontsize=14);
# plt.tight_layout();
_ = x=df['color'].value_counts().values

CPU times: user 76.9 ms, sys: 64 µs, total: 77 ms
Wall time: 76.4 ms


# Which Color is most popular among genders?

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 59 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 59 ##
cudf_df['Category'] = df['Category']

In [69]:
%%time
### cell 59 ###

# Popular Color in Kids Category

df.groupby(['Category','color']).size()['Kids']

CPU times: user 134 ms, sys: 48.3 ms, total: 183 ms
Wall time: 181 ms


color
Black         20000
Blue          12000
Green          4000
Grey           9000
Multicolor     1000
Orange         1000
Pink          13000
Purple         3000
Red            5000
White         26000
Yellow         3000
dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 60 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 60 ##
cudf_df['Category'] = df['Category']

In [70]:
%%time
### cell 60 ###

# Popular Color in Womens Category

df.groupby(['Category','color']).size()['Women']

CPU times: user 128 ms, sys: 52.2 ms, total: 181 ms
Wall time: 180 ms


color
Beige          1000
Black         67000
Blue          28000
Burgundy       5000
Gold           1000
Green         26000
Grey          23000
Multi          3000
Multicolor    10000
Orange         2000
Pink          44000
Purple        26000
Red           10000
Turquoise      1000
White         92000
Yellow         3000
dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 61 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 61 ##
cudf_df['Category'] = df['Category']

In [71]:
%%time
### cell 61 ###

# Popular Color in Mens Category

df.groupby(['Category','color']).size()['Men']

CPU times: user 157 ms, sys: 28.1 ms, total: 185 ms
Wall time: 184 ms


color
Beige          5000
Black         61000
Blue          47000
Burgundy       4000
Gold           2000
Green         22000
Grey          39000
Multi          1000
Multicolor     7000
Orange         7000
Pink           1000
Purple         2000
Red            8000
White         67000
Yellow         6000
dtype: int64

#### The top color in **Mens** category is '**White**' with '67' sales,the top color in **Womens** category is '**White**' with '92' sales and the top color in **Kids** category is '**White**' with '26' sales

# Availability

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 62 ##
cudf_df['availability'] = df['availability']

In [72]:
%%time
### cell 62 ###

# Unique Availability values in the DataFrame

df['availability'].unique()

CPU times: user 38.1 ms, sys: 0 ns, total: 38.1 ms
Wall time: 37.6 ms


array(['InStock', 'OutOfStock'], dtype=object)

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 63 ##
cudf_df['availability'] = df['availability']

In [73]:
%%time
### cell 63 ###

# No.of Unique Availability values in the DataFrame

df['availability'].nunique()

CPU times: user 34.2 ms, sys: 3.44 ms, total: 37.7 ms
Wall time: 37.1 ms


2

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 64 ##
cudf_df['availability'] = df['availability']

In [74]:
%%time
### cell 64 ###

# No.of Products according to Avaialability

df['availability'].value_counts()

CPU times: user 36.4 ms, sys: 28 µs, total: 36.4 ms
Wall time: 36.2 ms


availability
InStock       826000
OutOfStock      3000
Name: count, dtype: int64

#### There are '**826**' products which are '**InStock**' and '**3**' are '**OutOfStock**'.

## Product Type

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 65 ##
cudf_df['Product_Type'] = df['Product_Type']

In [75]:
%%time
### cell 65 ###

# Unique Product Type values in the DataFrame

df['Product_Type'].unique()

CPU times: user 111 ms, sys: 0 ns, total: 111 ms
Wall time: 110 ms


array(['Clothing', 'Shoes', 'Accessories'], dtype=object)

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 66 ##
cudf_df['Product_Type'] = df['Product_Type']

In [76]:
%%time
### cell 66 ###

# No.of Unique Product Type values in the DataFrame

df['Product_Type'].nunique()

CPU times: user 109 ms, sys: 58 µs, total: 109 ms
Wall time: 108 ms


3

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 67 ##
cudf_df['Product_Type'] = df['Product_Type']

In [77]:
%%time
### cell 67 ###

# No.of Products according to Product Type

df['Product_Type'].value_counts()

CPU times: user 95.3 ms, sys: 287 µs, total: 95.6 ms
Wall time: 94.7 ms


Product_Type
Shoes          422000
Clothing       328000
Accessories     79000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 68 ##
cudf_df['Product_Type'] = df['Product_Type']

In [78]:
%%time
### cell 68 ###

# No.of Products according to Product Type

# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['Product_Type'].value_counts().plot(kind='barh', color={'#864879','#2D4263','#C84B31'})
# plt.ylabel("Product Type")
# plt.xlabel("No.of Products Sold According to Particular - Product Type")
# plt.title("Product Type Vs. No.of Products Sold According to Particular - Product Type")
# plt.show();
df['Product_Type'].value_counts()

CPU times: user 82 ms, sys: 0 ns, total: 82 ms
Wall time: 81.5 ms


Product_Type
Shoes          422000
Clothing       328000
Accessories     79000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 69 ##
cudf_df['Product_Type'] = df['Product_Type']

In [79]:
%%time
### cell 69 ###

# No.of Products according to Product Type

labels = df['Product_Type'].value_counts().index
# STEFANOS: Disable plotting
# pie, ax = plt.subplots(figsize=[10,4])
# plt.pie(x=df['Product_Type'].value_counts().values, autopct="%.1f%%", explode=[0.05]*3, labels=labels, pctdistance=0.5)
# plt.title("No.of Products according to Product Type in the DataFrame", fontsize=14);
# plt.tight_layout();
_ = df['Product_Type'].value_counts().values

CPU times: user 160 ms, sys: 0 ns, total: 160 ms
Wall time: 159 ms


## Maximum Selling Price for Each Product Type

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 70 ##
cudf_df['Product_Type'] = df['Product_Type']

In [ ]:
%%time
## Transfer_pre 70 ##
cudf_df['Category'] = df['Category']

In [80]:
%%time
### cell 70 ###

df.groupby(['Product_Type', 'Category']).max()['selling_price']

CPU times: user 441 ms, sys: 28.1 ms, total: 469 ms
Wall time: 468 ms


Product_Type  Category  
Accessories   Kids           13
              Men            23
              Originals     196
              Running        21
              Soccer         32
              Training       36
              Women          40
Clothing      Essentials     44
              Kids           60
              Men           240
              Originals      60
              Sportswear     72
              Women         128
Shoes         Essentials     60
              Kids          128
              Men           180
              Originals     180
              Running       144
              Soccer        200
              Swim           48
              Women         126
Name: selling_price, dtype: int64

## Minimum Selling Price for Each Product Type

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 71 ##
cudf_df['Product_Type'] = df['Product_Type']

In [ ]:
%%time
## Transfer_pre 71 ##
cudf_df['Category'] = df['Category']

In [81]:
%%time
### cell 71 ###

df.groupby(['Product_Type', 'Category']).min()['selling_price']

CPU times: user 428 ms, sys: 31.6 ms, total: 459 ms
Wall time: 459 ms


Product_Type  Category  
Accessories   Kids          13
              Men           10
              Originals      9
              Running       21
              Soccer        18
              Training      12
              Women         10
Clothing      Essentials    44
              Kids          19
              Men           18
              Originals     60
              Sportswear    72
              Women         16
Shoes         Essentials    60
              Kids          24
              Men           20
              Originals     44
              Running       98
              Soccer        64
              Swim          20
              Women         20
Name: selling_price, dtype: int64

## Average Selling Price for Each Product Type

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 72 ##
cudf_df['Product_Type'] = df['Product_Type']

In [ ]:
%%time
## Transfer_pre 72 ##
cudf_df['Category'] = df['Category']

In [82]:
%%time
### cell 72 ###

df.groupby(['Product_Type', 'Category']).mean(numeric_only=True)['selling_price']

CPU times: user 249 ms, sys: 28.1 ms, total: 277 ms
Wall time: 276 ms


Product_Type  Category  
Accessories   Kids           13.000000
              Men            16.000000
              Originals      43.727273
              Running        21.000000
              Soccer         25.000000
              Training       21.360000
              Women          19.500000
Clothing      Essentials     44.000000
              Kids           30.555556
              Men            39.760000
              Originals      60.000000
              Sportswear     72.000000
              Women          38.924855
Shoes         Essentials     60.000000
              Kids           50.956522
              Men            79.676056
              Originals      89.400000
              Running       119.200000
              Soccer         84.000000
              Swim           32.142857
              Women          61.405229
Name: selling_price, dtype: float64

#### The Product Type with most sales is '**Shoes**', and Maximum Selling Price of '**240$**' according to Selling Price. 

# Category

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 73 ##
cudf_df['Category'] = df['Category']

In [83]:
%%time
### cell 73 ###

# Unique Category values in the DataFrame

df['Category'].unique()

CPU times: user 80.6 ms, sys: 7.31 ms, total: 87.9 ms
Wall time: 86.2 ms


array(['Women', 'Soccer', 'Men', 'Kids', 'Essentials', 'Originals',
       'Training', 'Swim', 'Running', 'Sportswear'], dtype=object)

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 74 ##
cudf_df['Category'] = df['Category']

In [84]:
%%time
### cell 74 ###

# No.of Unique Category values in the DataFrame

df['Category'].nunique()

CPU times: user 85.3 ms, sys: 87 µs, total: 85.4 ms
Wall time: 84.8 ms


10

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 75 ##
cudf_df['Category'] = df['Category']

In [85]:
%%time
### cell 75 ###

# No.of Products according to Category

df['Category'].value_counts()

CPU times: user 78.6 ms, sys: 215 µs, total: 78.8 ms
Wall time: 78.4 ms


Category
Women         342000
Men           279000
Kids           97000
Originals      58000
Training       25000
Soccer         12000
Swim            7000
Running         6000
Essentials      2000
Sportswear      1000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 76 ##
cudf_df['Category'] = df['Category']

In [86]:
%%time
### cell 76 ###

# Top-5 Products according to Category

# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['Category'].value_counts().head(5).plot(kind='barh', color={'#864879','#2D4263','#C84B31', '#ECDBBA', '#B3541E'})
# plt.ylabel("Category")
# plt.xlabel("No.of Products Sold According to Particular - Category")
# plt.title("Top 5 - Products Sold according to Category in the DataFrame")
# plt.show();
df['Category'].value_counts().head(5)

CPU times: user 66 ms, sys: 0 ns, total: 66 ms
Wall time: 65.3 ms


Category
Women        342000
Men          279000
Kids          97000
Originals     58000
Training      25000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 77 ##
cudf_df['Category'] = df['Category']

In [87]:
%%time
### cell 77 ###

# Top-5 Products according to Category

labels = df['Category'].value_counts().head(5).index
# STEFANOS: Disable plotting
# pie, ax = plt.subplots(figsize=[10,4])
# plt.pie(x=df['Category'].value_counts().head(5).values, autopct="%.1f%%", explode=[0.05]*5, labels=labels, pctdistance=0.5)
# plt.title("Top 5 - Products Sold according to Category in the DataFrame", fontsize=14);
# plt.tight_layout();
_ = df['Category'].value_counts().head(5).values

CPU times: user 137 ms, sys: 60 µs, total: 137 ms
Wall time: 136 ms


#### The 'Category' with highest no.of Products Sold is '**Women**' with '**342 Products**', followed by 'Men' and 'Women'.

# Average Rating

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 78 ##
cudf_df['average_rating'] = df['average_rating']

In [88]:
%%time
### cell 78 ###

# Unique 'Average Rating'in the DataFrame

df['average_rating'].unique()

CPU times: user 3.69 ms, sys: 7.63 ms, total: 11.3 ms
Wall time: 10.9 ms


array([4.8, 4.2, 4.4, 4.5, 4.6, 5. , 4.7, 4.3, 4. , 4.9, 3.9, 3.7, 3. ,
       4.1, 1. , 3.8])

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 79 ##
cudf_df['average_rating'] = df['average_rating']

In [89]:
%%time
### cell 79 ###

# No.of Unique 'Average Rating'in the DataFrame

df['average_rating'].nunique()

CPU times: user 8.45 ms, sys: 39 µs, total: 8.49 ms
Wall time: 7.68 ms


16

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 80 ##
cudf_df['average_rating'] = df['average_rating']

In [90]:
%%time
### cell 80 ###

# No.of time particular Average Rating Provided

df['average_rating'].value_counts()

CPU times: user 5.88 ms, sys: 3.82 ms, total: 9.7 ms
Wall time: 9.24 ms


average_rating
4.8    161000
4.7    159000
4.6    120000
4.5     99000
5.0     70000
4.2     51000
4.4     49000
4.9     44000
4.3     29000
3.9     14000
4.1     14000
4.0     12000
3.7      4000
3.0      1000
1.0      1000
3.8      1000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 81 ##
cudf_df['average_rating'] = df['average_rating']

In [91]:
%%time
### cell 81 ###

# Top 5 - Average Rating Provided

# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['average_rating'].value_counts().head(5).plot(kind='barh', color={'#864879','#2D4263','#C84B31', '#ECDBBA', '#B3541E'})
# plt.ylabel("Average Rating")
# plt.xlabel("No.of times particular rating provided")
# plt.title("Top 5 - Average Rating Provided in the DataFrame")
# plt.show();
df['average_rating'].value_counts().head(5)

CPU times: user 8.95 ms, sys: 0 ns, total: 8.95 ms
Wall time: 8.72 ms


average_rating
4.8    161000
4.7    159000
4.6    120000
4.5     99000
5.0     70000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 82 ##
cudf_df['average_rating'] = df['average_rating']

In [92]:
%%time
### cell 82 ###

# Top 5 - Average Rating Provided

labels = df['average_rating'].value_counts().head(5).index
# STEFANOS: Disable plotting
# pie, ax = plt.subplots(figsize=[10,4])
# plt.pie(x=df['average_rating'].value_counts().head(5).values, autopct="%.1f%%", explode=[0.05]*5, labels=labels, pctdistance=0.5)
# plt.title("# Top 5 - Average Rating Provided in the DataFrame", fontsize=14);
# plt.tight_layout();
_ = df['average_rating'].value_counts().head(5).values

CPU times: user 18 ms, sys: 182 µs, total: 18.2 ms
Wall time: 17.5 ms


In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 83 ##
cudf_df['average_rating'] = df['average_rating']

In [93]:
%%time
### cell 83 ###

# Maxmimum Average Rating

df['average_rating'].max()

CPU times: user 1.76 ms, sys: 0 ns, total: 1.76 ms
Wall time: 1.41 ms


5.0

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 84 ##
cudf_df['average_rating'] = df['average_rating']

In [94]:
%%time
### cell 84 ###

# Minimum Average Rating

df['average_rating'].min()

CPU times: user 2.62 ms, sys: 0 ns, total: 2.62 ms
Wall time: 1.69 ms


1.0

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 85 ##
cudf_df['average_rating'] = df['average_rating']

In [95]:
%%time
### cell 85 ###

# Mean Average Rating

round(df['average_rating'].mean(),2)

CPU times: user 2.59 ms, sys: 0 ns, total: 2.59 ms
Wall time: 1.69 ms


4.61

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 86 ##
cudf_df['Product_Type'] = df['Product_Type']

In [ ]:
%%time
## Transfer_pre 86 ##
cudf_df['Category'] = df['Category']

In [96]:
%%time
### cell 86 ###

# Maximum Average Rating across 'Product Type' and 'Category'

df.groupby(['Product_Type', 'Category']).max()['average_rating']

CPU times: user 430 ms, sys: 36.2 ms, total: 467 ms
Wall time: 465 ms


Product_Type  Category  
Accessories   Kids          4.9
              Men           5.0
              Originals     5.0
              Running       4.7
              Soccer        4.8
              Training      5.0
              Women         5.0
Clothing      Essentials    4.7
              Kids          5.0
              Men           5.0
              Originals     5.0
              Sportswear    4.8
              Women         5.0
Shoes         Essentials    4.2
              Kids          5.0
              Men           5.0
              Originals     4.8
              Running       4.8
              Soccer        4.7
              Swim          4.7
              Women         5.0
Name: average_rating, dtype: float64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 87 ##
cudf_df['Product_Type'] = df['Product_Type']

In [ ]:
%%time
## Transfer_pre 87 ##
cudf_df['Category'] = df['Category']

In [97]:
%%time
### cell 87 ###

# Minimum Average Rating across 'Product Type' and 'Category'

df.groupby(['Product_Type', 'Category']).min()['average_rating']

CPU times: user 411 ms, sys: 40.3 ms, total: 451 ms
Wall time: 450 ms


Product_Type  Category  
Accessories   Kids          4.9
              Men           4.7
              Originals     4.5
              Running       4.7
              Soccer        4.8
              Training      4.6
              Women         4.6
Clothing      Essentials    4.7
              Kids          4.0
              Men           1.0
              Originals     5.0
              Sportswear    4.8
              Women         3.7
Shoes         Essentials    4.2
              Kids          4.3
              Men           3.9
              Originals     3.9
              Running       4.6
              Soccer        4.3
              Swim          4.4
              Women         3.8
Name: average_rating, dtype: float64

#### The **Maximum** Average Rating among all products is '**5.0**' and **Minimum** Average Rating among all products is '**1.0**'

# Reviews Count

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 88 ##
cudf_df['reviews_count'] = df['reviews_count']

In [98]:
%%time
### cell 88 ###

# No.of Unique 'Reviews Count' in the DataFrame

df['reviews_count'].nunique()

CPU times: user 8.9 ms, sys: 81 µs, total: 8.98 ms
Wall time: 8.12 ms


211

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 89 ##
cudf_df['reviews_count'] = df['reviews_count']

In [99]:
%%time
### cell 89 ###

# Top 10 - 'Reviews Count' and no.of particular 'Reviews Count' occurances

df['reviews_count'].value_counts().head(10)

CPU times: user 4.45 ms, sys: 41 µs, total: 4.49 ms
Wall time: 4.38 ms


reviews_count
1      23000
2      21000
5      15000
9      15000
4      14000
8      14000
3      13000
562    12000
51     12000
454    12000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 90 ##
cudf_df['average_rating'] = df['average_rating']

In [100]:
%%time
### cell 90 ###

# Top 5 - 'Reviews Count' and no.of particular 'Reviews Count' occurances

# STEFANOS: Disable plotting
# plt.figure(figsize=(12,4))
# df['average_rating'].value_counts().head(5).plot(kind='barh', color={'#864879','#2D4263','#C84B31', '#ECDBBA', '#B3541E'})
# plt.ylabel('Reviews Count')
# plt.xlabel('no.of particular Reviews Count occurances')
# plt.title("Top 5 - 'Reviews Count' and no.of particular 'Reviews Count' occurances")
# plt.show();
df['average_rating'].value_counts().head(5)

CPU times: user 8.91 ms, sys: 79 µs, total: 8.99 ms
Wall time: 8.72 ms


average_rating
4.8    161000
4.7    159000
4.6    120000
4.5     99000
5.0     70000
Name: count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 91 ##
cudf_df['reviews_count'] = df['reviews_count']

In [101]:
%%time
### cell 91 ###

# Maxmimum 'Reviews Count'

df['reviews_count'].max()

CPU times: user 1.43 ms, sys: 89 µs, total: 1.52 ms
Wall time: 690 µs


11750

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 92 ##
cudf_df['reviews_count'] = df['reviews_count']

In [102]:
%%time
### cell 92 ###

# Minimum 'Reviews Count'

df['reviews_count'].min()

CPU times: user 1.34 ms, sys: 123 µs, total: 1.46 ms
Wall time: 675 µs


1

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 93 ##
cudf_df['reviews_count'] = df['reviews_count']

In [103]:
%%time
### cell 93 ###

# Mean 'Reviews Count'

round(df['reviews_count'].mean(),2)

CPU times: user 1.96 ms, sys: 1e+03 ns, total: 1.96 ms
Wall time: 1.21 ms


433.94

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 94 ##
cudf_df['Product_Type'] = df['Product_Type']

In [ ]:
%%time
## Transfer_pre 94 ##
cudf_df['Category'] = df['Category']

In [104]:
%%time
### cell 94 ###

# Maximum 'Reviews Count' across 'Product Type' and 'Category'

df.groupby(['Product_Type', 'Category']).max()['reviews_count']

CPU times: user 431 ms, sys: 44.5 ms, total: 476 ms
Wall time: 474 ms


Product_Type  Category  
Accessories   Kids             41
              Men             120
              Originals       298
              Running          25
              Soccer           38
              Training        273
              Women           154
Clothing      Essentials       29
              Kids             74
              Men            1352
              Originals         2
              Sportswear       26
              Women           671
Shoes         Essentials      120
              Kids            602
              Men           11750
              Originals     11750
              Running         584
              Soccer          281
              Swim           7291
              Women          9636
Name: reviews_count, dtype: int64

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 95 ##
cudf_df['Product_Type'] = df['Product_Type']

In [ ]:
%%time
## Transfer_pre 95 ##
cudf_df['Category'] = df['Category']

In [105]:
%%time
### cell 95 ###

# Minimum Reviews Count' across 'Product Type' and 'Category'

df.groupby(['Product_Type', 'Category']).min()['reviews_count']

CPU times: user 426 ms, sys: 31.8 ms, total: 458 ms
Wall time: 457 ms


Product_Type  Category  
Accessories   Kids           41
              Men             2
              Originals       1
              Running        25
              Soccer          5
              Training        1
              Women           2
Clothing      Essentials     29
              Kids            1
              Men             1
              Originals       2
              Sportswear     26
              Women           1
Shoes         Essentials    120
              Kids            3
              Men             1
              Originals      49
              Running       491
              Soccer         24
              Swim           79
              Women           3
Name: reviews_count, dtype: int64

#### The **Maximum** 'Reviews Count' among all products is '**11750**' and **Minimum** 'Reviews Count' among all products is '**1**'

# Reviews Count Vs. Average Rating

In [106]:
# STEFANOS: Disable plotting
# sns.scatterplot(data=df, x='average_rating', y='reviews_count')

In [ ]:
cudf_df = cudf.DataFrame()

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['name'] = df['name']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['selling_price'] = df['selling_price']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['original_price'] = df['original_price']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['availability'] = df['availability']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['color'] = df['color']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['average_rating'] = df['average_rating']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['reviews_count'] = df['reviews_count']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['Category'] = df['Category']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['Product_Type'] = df['Product_Type']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['Discount'] = df['Discount']

In [ ]:
%%time
## Transfer_pre 96 ##
cudf_df['Discount(%)'] = df['Discount(%)']

In [107]:
%%time
### cell 96 ###

df.shape

CPU times: user 13 µs, sys: 0 ns, total: 13 µs
Wall time: 15.5 µs


(829000, 11)

#### The Correlation Between Reviews Count & Average Rating is '**0.024**' (positive correlation), from the graph we may can conclude that there are more number of reviews when particular product rated above **3.5** 

## End of the Notebook, Thanks for Watching, "**Do Upvote**" if it is helpful.